FRAuth/FRAuth/Callbacks/IdPCallback.swift

```diff
@@ -0,0 +1,207 @@
+// 
+//  IdPCallback.swift
+//  FRAuth
+//
+//  Copyright (c) 2021 ForgeRock. All rights reserved.
+//
+//  This software may be modified and distributed under the terms
+//  of the MIT license. See the LICENSE file for details.
+//
+
+import Foundation
+import UIKit
+
+
+/**
+ IdPCallback is a representation of `Social Provider Handler Node` in AM when `Client Type` option specified as `NATIVE` (only available in AM 7.1 and above)
+ */
+public class IdPCallback: MultipleValuesCallback {
```

Comment:
Can we make this conform to Codable in order to make the JSON serialisation a bit cleaner/easier

Reply:
That's a good idea. I believe we could make the ```Callback``` class ```Codable```, which would be inherited by all callback classes

Reply:
@george-bafaloukas-forgerock , @rodrigoareis , I do agree that it would be much cleaner if we were to do it with Codable; however, if I recall correctly, I had tried with Codable approach when I initially implement Callback structure, and encountered some issue with Objc compatibility or something else. It is still worthwhile to investigate further on that topic, but as this would take entire Callback structure changes, are you guys okay with having a separate story for the investigation?

Thanks,

Reply:
Hey James, I do agree that this would require a significant amount of time to refactor. A spike to investigate how much this time would be, if we have any issue and what other classes would benefit from such refactoring is a good idea in my opinion